### DISCUSSION TITLE
shareinto: Add more tokens for "Share Into - Format" setting

### DISCUSSION THREAD

**Discussion by mglh** [gsantner/markor#1596] [Ideas]
## General information
* App version: v2.8.4
* Android version: 9
* Device: Sony Xperia XA2
* I understand that this issue will be closed without further notice in case I didn't describe the problem including general information, or didn't [search](https://github.com/gsantner/markor/issues?q=) for similar issues & [requests](https://github.com/gsantner/markor/discussions?discussions_q=).

### Description

AFAIU, currently the "Share Into - Format" is limited to generate Markdown formated links.

To overcome this, I would suggest to replace the `'%s'` wildcard with this ones:

* `'%name'`
* `'%url'`
* `'%newline'`

Admission: I assume that this would require a separate "Share Into Format" for each text format (Markdown, todo.txt, Zim Wiki, Plain Text).

**Comment by gsantner** [comment]
> Admission: I assume that this would require a separate "Share Into Format" for each text format (Markdown, todo.txt, Zim Wiki, Plain Text).

Ahead: This should be really really simple. I don't want that users need to type separate "Share Into Formats" for every file. Or bloat settings to make everything-to-everything configurable, requiring an editor for each.

**Reply by GlassGruber** [reply]
Can snippets be inserted in the Shareinto view? This could easily solve the token multi file config issue by just letting users create snippets and insert them as they like. 

Also, maybe it could be possible to call a snippet file via some `%` escaping? Like if snippet file name is `mysnippet.txt` something like `%mysnippet.txt%` will place it in the final output?

**Comment by gsantner** [comment]
You can directly write what you want there in the shareinto format.

This should stay simple and not get bloated with everything.

**Reply by GlassGruber** [reply]
My bad, what I meant is not inside the format option, I was talking about the main Shareinto view, where you are presented with the content of what is going to be shared to a document. But anyway I add the snippet after I opened the document I intended to share the text to. So just a few more taps away and yes better keep it lean and clean, still I'm a lazy bum that is always looking for less taps :P .

I take the chance to thank you for this awesome app! A must install ! 🎩 🤖 

![image](https://user-images.githubusercontent.com/1172428/172279129-ba775305-57a5-411b-8c94-860e6371381c.png)
